tests/Twig/Extension/UnicodeStringTest.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\AdminBundle\Tests\Twig\Extension;
+
+use PHPUnit\Framework\TestCase;
+use Sonata\AdminBundle\Twig\Extension\UnicodeString;
+use Symfony\Component\String\Exception\InvalidArgumentException;
+
+/**
+ * @author Javier Spagnoletti <phansys@gmail.com>
+ */
+final class UnicodeStringTest extends TestCase
+{
+    public function testCreateFromStringWithInvalidUtf8Input(): void
+    {
+        $this->expectException(InvalidArgumentException::class);
+
+        static::createFromString("\xE9");
+    }
+
+    public function testAscii(): void
+    {
+        $s = static::createFromString('Dieser Wert sollte größer oder gleich');
+        $this->assertSame('Dieser Wert sollte grosser oder gleich', (string) $s->ascii());
+        $this->assertSame('Dieser Wert sollte groesser oder gleich', (string) $s->ascii(['de-ASCII']));
+    }
+
+    /**
+     * @dataProvider provideTruncate
+     */
+    public function testTruncate(string $expected, string $origin, int $length, string $ellipsis, bool $preserve = false): void
+    {
+        $instance = static::createFromString($origin)->truncate($length, $ellipsis, $preserve);
+
+        $this->assertSame((string) static::createFromString($expected), (string) $instance);
+    }
+
+    public static function provideTruncate(): iterable
+    {
+        yield from [
```

Comment:
What's the point of using this instead of `return`, or several `yield`?

Reply:
IMO, think the point is avoid providing a whole array to the test method, and instead, provide its values one by one (regardless the provider was already reserved the memory for the array).
Please, note that I don't know the internals about this, so I don't know if the GC is able to leverage this construction in some way.

Reply:
I see, so it's less memory efficient than several `yield`, but more than a `return` and it's easier to write than several `yield`

Reply:
I think yes.
I don't know if it's more efficient since the memory is surely already reserved when the array is created; but I guess at least, it shouldn't pass the whole array to the PHPUnit method responsible to provide the data for the test method.